tensorflow/stream_executor/rocm/rocm_dnn.cc

```diff
@@ -3104,23 +3104,55 @@ port::Status MIOpenSupport::DoConvolve(
     }
   }
 
+  std::unique_ptr<TemporaryDeviceMemory<uint8>> local_scratch;
+  uint8* scratch_memory_ptr = reinterpret_cast<uint8*>(scratch_memory.opaque());
+  size_t scratch_memory_size = scratch_memory.size();
+
   miopenStatus_t status = miopenStatusSuccess;
+  size_t required_scratch_memory_size=0;
+
+    // On the first call with XLA without autotune, required scratch size is 
+    // unknown, and we're getting the ScratchAllocator with internal capacity
+    // zero. Therefore we have to fall back on alternate means to allocate
+    // any scratch memory actually needed.
+  if(kind == dnn::ConvolutionKind::FORWARD)
+    status = wrap::miopenConvolutionForwardGetWorkSpaceSize(miopen.handle(), 
```

Comment:
At present, the switch between find mode and immediate mode is done by environment variable. Switching depending on whether autotune was run would be a lot of architectural work. Since the tests pass anyway, I don't see an urgent need to do that now (we could put it on the to-do list...)

Reply:
there are a couple of things we need to flush out with this implementation
 * it unconditionally calls the *GetWorkSpaceSize routine everytime we do a convolution...need to check with the MIOpen team how expensive/cheap this call is...it may be adding non-significant overhead
* In Find Mode the value returned by *GetWorkSpaceSize is the worst-case number for a given conv, which will always be >= the actual scratch space size needed by the algorithm picked by the auto-tuner. So in the normal scenario where we run the auto-tuner, this implementation will end up allocating extra scratch memory in those cases!

Reply:
Should we revert this for now? I don't have time to fix it right away and I'm out next week.

Reply:
Yes, revert for now, and then we can get to it later.

in theory, this should only affect the case `*autotune_disabled` testcase....wondering why it affects the other two

Reply:
The failing subtest of /tensorflow/compiler/xla/tests:convolution_test_gpu disables autotune as well.